File: BackendAPI/Application/DTOs/Output/OrderOutputDto.cs
using Domain.Enums;

namespace Application.DTOs.Output;

public class OrderOutputDto
{
    public required Guid Id { get; set; }
    public required Guid ProductId { get; set; }
    public required string ProductName { get; set; }
    public required string ProductDescription { get; set; }
    public required string ProductImageUrl { get; set; }
    public required string CompanyName { get; set; }
    public required DateTimeOffset CreatedAt { get; set; }
    public required OrderStatus Status { get; set; }
    public required DateTimeOffset? ClosedAt { get; set; }
    public required decimal TotalAmount { get; set; }
    public required int TotalItems { get; set; }
}

File: FrontendWeb/src/declarations/dtos/output/OrderOutputDto.ts
import { OrderStatus } from '../../enums/OrderStatus';

// OrderOutputDto.ts
export interface OrderOutputDto {
	id: string;
	createdAt: string;
	status: OrderStatus;
	closedAt?: string;
	totalAmount: number;
	totalItems: number;
	productId: string;
	productName: string;
	productDescription: string;
	productImageUrl: string;
	companyName: string;
}
